File: src/utils/cache.js
/*
 * @Description: 
 * @Author: sky
 * @Date: 2024-07-23 17:20:19
 * @LastEditTime: 2025-01-07 08:44:46
 * @LastEditors: sky
 */
// cache.js
class LocalCache {
    constructor(prefix = '') {
      this.prefix = prefix;
    }
  
    setItem(key, value) {
      try {
        const serializedValue = JSON.stringify(value);
        localStorage.setItem(this.prefix + key, serializedValue);
      } catch (error) {
        console.error('Error setting item in local storage:', error);
      }
    }
  
    getItem(key) {
      try {
        const serializedValue = localStorage.getItem(this.prefix + key);
        if (serializedValue === null) {
          return null;
        }
        return JSON.parse(serializedValue);
      } catch (error) {
        console.error('Error getting item from local storage:', error);
        return null;
      }
    }
  
    removeItem(key) {
      localStorage.removeItem(this.prefix + key);
    }
  
    clear() {
      const keys = Object.keys(localStorage)
                        .filter(k => k.startsWith(this.prefix))
                        .forEach(k => localStorage.removeItem(k));
    }

    // 获取所有缓存数据
    getAllItems() {
          const allItems = [];
          Object.keys(localStorage).forEach(key => {
            if (key.startsWith(this.prefix)) {
              const value = this.getItem(key.replace(this.prefix, ''));
              allItems.push(value);
            }
          });
          return allItems;
    }
  }
  

  const cache = new LocalCache('SkyAnime-');
  export default cache;

// 使用示例
//   cache.setItem('user', { name: 'John Doe', age: 30 });
//   const user = cache.getItem('user');
//   console.log(user); // 输出: { name: 'John Doe', age: 30 }
  
//   cache.removeItem('user');
//   cache.clear();

File: app.bak/parser/scrapy_tv/scrapy_tv/cache.py
'''
Description: 缓存管理器
Author: sky
Date: 2024-07-04 08:59:08
LastEditTime: 2024-07-04 09:00:10
LastEditors: sky
'''
import diskcache

class MyCache:
    """缓存管理器"""

    def __init__(self, cache_path):
        self.cache = diskcache.Cache(cache_path)

    def set(self, key, value):
        """设置缓存"""
        self.cache.set(key, value)

    def get(self, key):
        """获取缓存"""
        return self.cache.get(key)
    
    def top(self, limit=50):
        """获取部分缓存,默认50条"""
        # 获取所有缓存项的键
        all_keys = self.cache.iterkeys()

        # 用于计数的变量和保存结果的字典
        count = 0
        items = {}
        for key in all_keys:
            value = self.cache.get(key)
            items[key] = value
            count += 1
            if count == int(limit):
                break

        return items